### DISCUSSION TITLE
Does ParU support multiple rhs (b as a multi-vector or arguably matrix)

### DISCUSSION THREAD

**Discussion by mzy2240** [DrTimothyAldenDavis/SuiteSparse#878] [Q&A]
Congratulations on releasing the stable version of ParU!
Just wondering if ParU supports multiple rhs. Feels OpenMP would make ParU very powerful on solving multi-rhs efficiently.

**Comment by DrTimothyAldenDavis** [comment]
Yes, it supports X=A\B where X and B are matrices (dense matrices to be precise, not sparse; only A is sparse).

We exploit the parallelism inside the BLAS for this case, which is better than exploiting it per column or groups of columns for X and B.

**Comment by mzy2240** [comment]
That sounds exciting! Is there a dedicated method for this use case? I have some hard time trying to understand how `ParU_Solve (Sym, Num, b, x, &Control)` would work with dense matrix B. How to provide necessary information like dimension of B to the solver?

**Comment by DrTimothyAldenDavis** [comment]
You use Paru_Solve (Sym, Num, nrhs, B, X, Control).
